pkg/services/internal/whatsonchain/woc_get_merkle_path.go

```diff
@@ -0,0 +1,161 @@
+package whatsonchain
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/bsv-blockchain/go-sdk/chainhash"
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	"github.com/go-resty/resty/v2"
+	"github.com/go-softwarelab/common/pkg/must"
+)
+
+// tscProof represents the response from the WoC /tx/{txid}/proof/tsc endpoint
+type tscProof struct {
+	Index  int      `json:"index"`
+	Nodes  []string `json:"nodes"`
+	Target string   `json:"target"` // block hash
+	TxOrID string   `json:"txOrId"` // txid
+}
+
+// blockHeaderResponse represents the response from WoC /block/{hash}/header
+type blockHeaderResponse struct {
+	Height     int    `json:"height"`
+	MerkleRoot string `json:"merkleroot"`
+}
+
+// MerklePath retrieves the merkle path for a transaction using WoC TSC proof.
+func (woc *WhatsOnChain) MerklePath(ctx context.Context, txID string) (*wdk.MerklePathResult, error) {
+	req := woc.httpClient.R().
+		SetContext(ctx).
+		SetHeader("Accept", "application/json").
+		AddRetryCondition(retryOn429)
```

Comment:
```suggestion
	var proof tscProof
	req := woc.httpClient.R().
		SetContext(ctx).
		SetResult(&proof).
		SetHeader("Accept", "application/json").
		AddRetryCondition(retryOn429)
```

This way you don't have to unmarshall the response object manually.